src/Compilers/CSharp/Portable/Lowering/SyntheticBoundNodeFactory.cs

```diff
@@ -1298,7 +1298,7 @@ public BoundExpression Convert(TypeSymbol type, BoundExpression arg)
 #endif 
             Conversion c = Compilation.Conversions.ClassifyConversionFromExpression(arg, type, isChecked: false, ref useSiteInfo);
             Debug.Assert(c.Exists);
-            Debug.Assert(useSiteInfo.Diagnostics.IsNullOrEmpty());
+            Debug.Assert(!useSiteInfo.HasErrors);
 
             return Convert(type, arg, c);
```

Comment:
>Convert [](http://example.com/codeflow?start=19&length=7)

Are we dropping use-site warnings in cases where the previous assertion failed?

Reply:
The `Convert` method doesn't report those use-site diagnostics (we use `DiscardedDependencies`/`Discarded` above).
In both scenarios, the use-site warnings are reported elsewhere.

Reply:
From our offline discussion, it sounds like the assert is checking we haven't reached lowering with an error.